File: cpp/3rd_Week/Praktikum/latihanpraktikum3_cpp/Pc.cpp
#include <iostream>
#include <string>
// #include "Processor.cpp"
// #include "Disk.cpp"
// #include "Ram.cpp"

class Pc
{
private:
    /* data */
    Processor procObj;
    Disk diskObj;
    Ram ramObj;
    long totalPrice;

public:
    Pc() {}
    Pc(Processor p, Disk d, Ram r) {
        this->procObj = p;
        this->diskObj = d;
        this->ramObj = r;
        // this->totalPrice = total;
    }

    void setProcObj(Processor p){ this->procObj = p; }
    void setDiskObj(Disk d){ this->diskObj = d; }
    void setRamObj(Ram r){ this->ramObj = r; }
    // void setTotalPrice(long total){ this->totalPrice = total; }

    Processor getProcObj(){ return this->procObj; }
    Disk getDiskObj(){ return this->diskObj; }
    Ram getRamObj(){ return this->ramObj; }
    // long getTotalPrice(){ return this->totalPrice; }

    // method
    void countTotal(){
        totalPrice = this->procObj.getProcPrice() + this->diskObj.getDiskPrice() + this->ramObj.getRamPrice();
    }

    void printPc(){
        this->procObj.printProcessor();
        this->diskObj.printDisk();
        this->ramObj.printRam();
        cout << "-Data PC" << endl;
        cout << " Harga PC : " << this->totalPrice << endl;
        cout << "-----------------------------------" << endl;
    }

    ~Pc() {}
};

File: java/javadpbo/src/com/sem4dpbo/week3/praktikum3/latihanpraktikum3_java/PC.java
package com.sem4dpbo.week3.praktikum3.latihanpraktikum3_java;

public class PC {
    private Processor procObj;
    private Disk diskObj;
    private Ram ramObj;
    private long totalPrice;

    PC(){}
    PC(Processor procObj, Disk diskObj, Ram ramObj){
        this.procObj = procObj;
        this.diskObj = diskObj;
        this.ramObj = ramObj;
    }

    public Processor getProcObj() {
        return procObj;
    }

    public void setProcObj(Processor procObj) {
        this.procObj = procObj;
    }

    public Disk getDiskObj() {
        return diskObj;
    }

    public void setDiskObj(Disk diskObj) {
        this.diskObj = diskObj;
    }

    public Ram getRamObj() {
        return ramObj;
    }

    public void setRamObj(Ram ramObj) {
        this.ramObj = ramObj;
    }

    public void countTotal(){
        totalPrice = this.procObj.getProcPrice() + this.diskObj.getDiskPrice() + this.ramObj.getRamPrice();
    }

    public void printPc(){
        this.procObj.printProcessor();
        this.diskObj.printDisk();
        this.ramObj.printRam();
        System.out.println("-Data Pc");
        System.out.println(" Total Price: " + this.totalPrice);
        System.out.println("------------------------------------");
    }
}
